### PR TITLE
Hdrp/refactor reflection probe/clean proxy volume

### PR DESCRIPTION
**Pull request by RSlysz**
Clean ProxyVolume classes and associated editors

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/HDRP/Lighting/Reflection/ProbeWrapper.cs [hunk 1]
```diff
@@ -27,6 +27,19 @@ protected static EnvShapeType ConvertShape(Shape shape)
             }
         }
 
+        protected static EnvShapeType ConvertShape(ShapeOrInfinite shape)
+        {
+            switch (shape)
+            {
+                default:
+                case ShapeOrInfinite.Infinite:
```
Comment (fredpointzero) [inline com.unity.render-pipelines.high-definition/HDRP/Lighting/Reflection/ProbeWrapper.cs]: Prefer to use the sphere shape for infinite. (Sphere is cheaper in shader code)

File: com.unity.render-pipelines.high-definition/HDRP/Lighting/Reflection/Volumes/ShapeType.cs [hunk 2]
```diff
@@ -5,4 +5,11 @@ public enum Shape
         Box,
         Sphere,
     }
+
+    public enum ShapeOrInfinite
```
Comment (fredpointzero) [inline com.unity.render-pipelines.high-definition/HDRP/Lighting/Reflection/Volumes/ShapeType.cs]: `ShapeOrInfinite` is a confusing name, I prefer `Shape`.

This is more confusing when reading core, it sounds more like a variable than a type name.

**Comment by RSlysz** [comment]
Shape is already used for InfluenceVolume's shape which only contrains Box
and Sphere value. Are InfluenceShape and ProxyShape more suitable?

>

**Comment by sebastienlagarde** [comment]
Intergate change from FRed, Verbal approve from FRed :)

**Pull request merged**